### PR TITLE
Remove unnecessary rooting from `CSSStyleDeclaration`

### PR DESCRIPTION
**Pull request by r-52**
This commit will replace calls to `self.owner.root()` with `self.owner` to avoid unnecessary rooting of JS elements objects.

Ref.- Issue: #8126

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8137)

### CODE REVIEW

**Comment by nox** [comment]
Great, thanks! Just one issue to address and your contribution can be merged.

-S-awaiting-review +S-needs-code-changes 


---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[components/script/dom/cssstyledeclaration.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1Atr88_Vqk-_d_COPc)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L84)):</sup>
No need to root it here either, AFAICT.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by romankl** [comment]
<sup>**[components/script/dom/cssstyledeclaration.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1Atr88_Vqk-_d_COPc)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L84)):</sup>
I left/ changed it because `self.owner` doesn't have the `dom::bindings::js::RootedReference` which is required for `window_from_node`/ `RootedReference`. Should I change it?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by nox** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[components/script/dom/cssstyledeclaration.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1Atr88_Vqk-_d_COPc)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L84)):</sup>
This is not `RootedReference::r`, that's just `Root::r`, an intrinsic method of `Root`. You can just pass `&self.owner` (just like you could pass `&root`, actually). Note how `RootedReference` is not implemented on `Root`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by romankl** [comment]
<sup>**[components/script/dom/cssstyledeclaration.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1Atr88_Vqk-_d_COPc)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L84)):</sup>
Passing `&self.owner` over isn't possible, because the traits `dom::bindings::utils::Reflectable` and dom::bindings::conversions::DerivedFrom<dom::node::Node>` are missing for `self.owner`. What am I missing?


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by nox** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[components/script/dom/cssstyledeclaration.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1Atr88_Vqk-_d_COPc)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L84)):</sup>
Oh, right. You need to explicitly deref it, or I have no idea what I'm doing: `window_from_node(&*self.owner)`.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

File: components/script/dom/cssstyledeclaration.rs [hunk 1]
```diff
@@ -182,8 +180,7 @@ impl CSSStyleDeclarationMethods for CSSStyleDeclaration {
         // Step 3
         } else {
             // FIXME: extra let binding https://github.com/rust-lang/rust/issues/22323
```
Comment (dzbarsky) [inline components/script/dom/cssstyledeclaration.rs]: Nix the FIXME?

**Comment by romankl** [comment]
<sup>**[components/script/dom/cssstyledeclaration.rs, line 84 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1Atr88_Vqk-_d_COPc)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L84)):</sup>
Thank you for the help!! Made the requested change


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by nox** [comment]
One final nit and it's good to go.

-S-awaiting-review +S-needs-code-changes 


---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[components/script/dom/cssstyledeclaration.rs, line 182 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1At4nrS-aVZalorY3h-r1-182)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L182)):</sup>
Yeah, that commit makes this obsolete, please remove it and I can r+ it. Thanks again for your contribution.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by romankl** [comment]
I've removed the comment


---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by nox** [comment]
You forgot one, thanks for keeping up with it!

-S-awaiting-review +S-needs-code-changes 


---

Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[components/script/dom/cssstyledeclaration.rs, line 182 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1At4nrS-aVZalorY3h-r1-182)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L182)):</sup>
The comment appears twice. :)


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by romankl** [comment]
<sup>**[components/script/dom/cssstyledeclaration.rs, line 182 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1At4nrS-aVZalorY3h-r1-182)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L182)):</sup>
Sorry that I missed it :/ thanks again for the review - remove the other one too


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Comment by nox** [comment]
@bors-servo r+


---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[components/script/dom/cssstyledeclaration.rs, line 182 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8137#-K1At4nrS-aVZalorY3h-r1-182)** ([raw file](https://github.com/servo/servo/blob/0bf76e51cc30f72a3b2f2d92d0a2780277e7cc9c/components/script/dom/cssstyledeclaration.rs#L182)):</sup>
No problem, that happens. LGTM.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8137)

**Pull request merged**